micrometer-core/src/main/java/io/micrometer/core/instrument/binder/okhttp3/OkHttpMetricsEventListener.java

```diff
@@ -114,13 +118,12 @@ private void time(CallState state) {
         String uri = state.response != null && (state.response.code() == 404 || state.response.code() == 301)
                 ? "NOT_FOUND" : urlMapper.apply(request);
 
-        Tags requestTags = requestAvailable ? getRequestTags(request) : Tags.empty();
+        Tags requestTags = requestAvailable ? getRequestTags(request).and(generateTagsForRoute(request)) : Tags.empty();
 
         Iterable<Tag> tags = Tags.concat(extraTags, Tags.of(
             "method", requestAvailable ? request.method() : "UNKNOWN",
             "uri", requestAvailable ? uri : "UNKNOWN",
-            "status", getStatusMessage(state.response, state.exception),
-            "host", requestAvailable ? request.url().host() : "UNKNOWN"
```

Comment:
I like the change and unifying tags, however, we need to include a way for users to opt out and return to the current way to remain backwards compatible with their dashboard, etc.

You could do it either by providing a way to configure the old behavior, or provide something like a `OkHttpHostTagFilter` that is a meter filter that users could use to switch the tag back.

Reply:
Fair point. When I pushed the code, it occurred to me I haven't rethought backwards compatibility. What is the preferred way for micrometer? To default to the backwards-compatible way and opt-in for new behaviour. Or the other way round: default to the preferred way and have a way to opt-out and fallback to the old approach? 

I'd happily work on that further.

Reply:
Nevermind, I've reread your post and it contains the answer ;) 
I'll create a way to opt-out and return to the existing way of tagging.

Reply:
I would rely on @shakuzen to make the call. But I believe he wants to save backwards incompatible changes for a 2.0 release.

So probably make it so the user would opt-in for the new behavior.